### DISCUSSION TITLE
API for publish forms?

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#4342] [Q&A]
I'm probably using entirely incorrect terminology in the heading of the question, but that's besides the point.

Basically, what I want to do is point to a url to create a particular type of entry (this part is easy), but I want one or more of the fields in the blueprint to be pre populated based on the url (GET variables?).

For example: `domain.com/cp/collections/pages/entries/create/default` is a link to create a new page entry, but no fields are pre populated.

I need something like this: `domain.com/cp/collections/pages/entries/create/default?title="foo"&date="2021-09-30"`

Is that possible?

**Comment by duncanmcclean** [comment]
You'd likely have to build your own fieldtype that extends the existing ones so you can set the value on page load.

**Reply by ARCHES-Academy** [reply]
Well, I was hoping to not have to dig so deep, but I guess I can give that a try. I've never actually made a fieldtype before, so I'll likely have to do some trial and error to get it right.

**Reply by ARCHES-Academy** [reply]
I'm looking through the date fieldtype (the one I'm most concerned with adding the functionality to), and I'm pretty darn confused as to what's going on.

I think I would just need to call the `handleUpdate(value)` function on initial page load, but I haven't the foggiest idea how to do that. As I've said, I've never actually created or even extended a fieldtype before.

When extending the fieldtype, I know that I should start with:
```
use Statamic\Fieldtypes\Date;

class GetDate extends Date {
```
and then I should have access to everything that Date has, but what do I put in the vue component? Should I copy & paste the original date component? Leave it empty or delete it? Sorry if these are dumb questions, I'm just trying to wrap my head around what's happening.

From there, I assume that I should override a method that will be called at page load, like `preProcessRange($value)`, and then pull the url variables in to replace the `$value` variable? Or would it be better to create a new method and trigger it at page load instead?
